posts/2021-08-13-dev-mode-multi-module.adoc

```diff
@@ -0,0 +1,130 @@
+
+In this blog post, we introduce Open Liberty dev mode support for working with multi-module Maven projects.  This provides hot deployment of changes in all modules of a Maven reactor build, on-demand and hot testing for multiple modules, and the ability to attach a debugger to debug code in any module.  
+
+Dev mode support for multi-module Maven projects is available starting in version `3.4` of the Liberty Maven plug-in, which is available now.
+
+== Multi-module projects
+
+With multi-module Maven projects, there is a top-level aggregator `pom.xml` file that manages a group of modules. This is convenient as you can use one single command to build the entire project, and the link:https://maven.apache.org/guides/mini/guide-multiple-modules.html[Maven reactor] will determine the correct build order for you.  
+
+The following example is the link:https://github.com/OpenLiberty/guide-maven-multimodules[Open Liberty multi-module sample application]. This project contains three modules: a Java library in the `jar` folder, a web module in the `war` folder, and template files and the Liberty server configuration in the `ear` folder. 
+
+----
+guide-maven-multimodules
+|_ pom.xml
+|_ jar
+    |_pom.xml
+    |_src/main/java
+|_ war
+    |_pom.xml
+    |_src/main
+        |_java
+        |_webapp
+|_ ear
+    |_pom.xml
+    |_src
+        |_main/liberty/config
+        |_test/java
+----
+
+Each module is defined in the top-level aggregator `pom.xml` file:
+
+[source,xml]
+----
+    <modules>
+        <module>jar</module>
+        <module>war</module>
+        <module>ear</module>
+    </modules>
+----
+
+The modules within the project can also have dependencies on each other. In this example, the `war` module depends on the `jar` module, and the `ear` module depends on both the `war` and `jar` modules. By leveraging dev mode for multiple modules, not only will the code changes you make be automatically picked up and hot deployed, but any dependent modules will also be recompiled and hot deployed.
+
+== Running Liberty on a multi-module project
+
+Liberty can be run on a multi-module project with a single command by using dev mode, dev mode with container support, or by starting the server in the foreground.  To do any of these, the Liberty Maven plug-in must first be defined.
+
+=== Defining the Liberty Maven plug-in
+
+To use the Liberty Maven plug-in in your multi-module project, you can define the plug-in in either the parent `pom.xml` file of every module or in the `pom.xml` of every module.
+
+For example, if your modules have a `<parent>` element that points to a parent project, add the Liberty Maven plug-in to the `<pluginManagement>` section of the parent project's `pom.xml` as follows:
```

Comment:
`<parent>` 
--->
`parent`

In the docs, we don't include brackets around xml elements in text descriptions. Ok if you think it's needed here but they are already described as "elements" so might be redundant to include the brackets too.

Reply:
I think it helps to include brackets here to make it clear that we are referring to specific tags within XML files.